### PR TITLE
Add a draw during show for macos backend

### LINKED ISSUE #27953: [Bug]: Pyplot can no longer set axes properties
### Bug summary

Since Matplotlib 3.8.1, it no longer is possible to set axes properties, neither using Pyplot free functions nor Axis object methods.

This persists in the current latest release 3.8.3, and is not observed in 3.8.0.

### Code for reproduction

```Python
import matplotlib.pyplot as plt
import numpy as np
plt.plot(np.random.randn(100))
plt.xlabel("x label")
plt.show()
```

### Actual outcome

The "x label" text is not shown. The result is identical to just calling `plt.plot(...)` and `plt.show()`.

### Expected outcome

The "x label" text is shown.

### Additional information

The above example applies to all functions that set axes properties.

I have tried in many different ways to create a figure with axes, none of them resulted in me being able to set properties.

### Operating system

macOS 14.3.1 (aarch64)

### Matplotlib Version

3.8.1

### Matplotlib Backend

MacOSX

### Python version

3.11.8

### Jupyter version

_No response_

### Installation

pip

### PR DESCRIPTION
**Pull request by ksunden**
Without this a stale figure is shown

## PR summary

This is basically the same concept as what happens just before show in [`plt.pause`](https://github.com/matplotlib/matplotlib/blob/1174aadafd9ae55818436bd3033b36626dcd3b0d/lib/matplotlib/pyplot.py#L731).

Essentially, there is nothing in the macos backend that actually causes a stale figure to be redrawn on `show`.
Other backends (qt, tk in particular, but likely others) seem to be saved by the fact that a resize event is triggered upon show, but macos does not do the same.

Other ideas considered, would be to do this in `plt.show` (which would perhaps leave out some other mechanisms, such as mpl-gui) or to try and do this in backend_bases (which is hard because most do not call `super`).
Could also leave out the condition and just always call `draw_idle`.

closes #27953

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by greglucas — APPROVED** [review]
Thanks for the pointer about this needing to be run in a terminal and not a script. I can reproduce the original issue that way.

This fixes the issue, so has my approval and is good to go in. But I am still not sure whether there is a deeper issue here with the way we are setting the interrupt handling that this is just covering up in a sense by forcing another event through.

In the `__init__` method, we are actually following the call to `self.show()` with a `draw_idle()`. I'm wondering if we can either remove that call now, or if we should just be calling `draw_idle()` here regardless of whether we are stale or not like above.

https://github.com/matplotlib/matplotlib/blob/1174aadafd9ae55818436bd3033b36626dcd3b0d/lib/matplotlib/backends/backend_macosx.py#L159-L161

**Comment by ksunden** [comment]
Ping for second review here, I'd like to get this in for 3.8.4, which I would like to get out soon for np2 reasons.

I'm not too concerned about potentially doubling up draw_idle requests here, as they are intentionally pretty cheap in the first place. I am more interested in getting the fix in than tracking down if other calls could possibly be omitted.

I think it is likely correct that we could omit the call in `__init__` (even with the stale check), but also think the performance concern is quite minimal and am not 100% sure. (I would be pretty sure if we omitted the stale check, but not clear to me which would be "better")

**Pull request merged**
